src/tracing/builder/geth.rs

```diff
@@ -347,4 +353,195 @@ impl<'a> GethTraceBuilder<'a> {
+                    if let Some(stack) = &step.stack {
+                        if let Some(slot) = stack.get(stack.len().saturating_sub(1)) {
+                            let slot: B256 = (*slot).into();
+                            *accessed_slots.transient_writes.entry(slot).or_insert(0) += 1;
+                        }
+                    }
+                }
+                _ => {}
+            }
+
+            // Out-of-Gas detection
+            if let Some(status) = &step.status {
+                if *status == revm::interpreter::InstructionResult::OutOfGas {
+                    out_of_gas = true;
+                }
+            }
+
+            // EXT opcode handling and contract size capture
+            if matches!(op, opcode::EXTCODESIZE | opcode::EXTCODECOPY | opcode::EXTCODEHASH) {
+                if let Some(stack) = &step.stack {
+                    if let Some(item) = stack.get(stack.len().saturating_sub(1)) {
+                        let address = Address::from(item.to_be_bytes());
+                        ext_code_access_info.push(format!("{address:?}"));
+                        if let Entry::Vacant(e) = contract_size.entry(address) {
+                            if let Ok(Some(account)) = db.basic_ref(address) {
+                                if let Ok(bytecode) = db.code_by_hash_ref(account.code_hash) {
+                                    let size = bytecode.len();
+                                    e.insert(ContractSize {
+                                        contract_size: size as u64,
+                                        opcode: op,
+                                    });
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            // KECCAK preimages from returndata
+            if op == opcode::KECCAK256 {
+                if let (Some(stack), Some(memory)) = (&step.stack, &step.memory) {
+                    if stack.len() >= 2 {
+                        let offset = stack[stack.len() - 1];
+                        let len = stack[stack.len() - 2];
+                        let offset_usize: usize = offset.try_into().unwrap_or(0);
+                        let len_usize: usize = len.try_into().unwrap_or(0);
+                        if offset_usize
+                            .checked_add(len_usize)
+                            .is_some_and(|end| end <= memory.0.len())
+                        {
+                            let data = memory.0[offset_usize..offset_usize + len_usize].to_vec();
+                            keccak.push(data);
+                        }
+                    }
+                }
+            }
+        }
+
+        let mut calls = Vec::new();
+        for (idx, n) in self.nodes.iter().enumerate() {
+            if n.parent == Some(node_idx) {
+                let child_trace =
+                    self.geth_erc7562_traces_internal(idx, opts.clone(), gas_used, &db);
```

Comment:
this is a recursive call, right?

this tree can be quite deep and can result in stack overflows, so we should solve this with a stack based approach instead, similar to callframe impl